### PR TITLE
Switch to golangci-lint

### PR DESCRIPTION
**Pull request by talal**

### CODE REVIEW

File: internal/api/registry/uploads.go [hunk 1]
```diff
@@ -417,8 +418,10 @@ func (a *API) handleFinishBlobUpload(w http.ResponseWriter, r *http.Request) {
 		contentLength, err := strconv.ParseUint(contentLengthStr, 10, 64)
 		if err != nil {
 			//COVERAGE: unreachable in unit tests because net/http validates Content-Length header format before sending
-			keppel.ErrSizeInvalid.With("malformed Content-Length: " + err.Error())
-			return
+			err = keppel.ErrSizeInvalid.With("malformed Content-Length: " + err.Error())
+			if respondWithError(w, r, err) {
+				return
+			}
```
Comment (majewsky) [inline internal/api/registry/uploads.go:424]: Nice catch from the lint. But the common pattern to report errors of concrete types is
```suggestion
			keppel.ErrSizeInvalid.With("malformed Content-Length: " + err.Error()).WriteAsRegistryV2ResponseTo(w, r)
			return
```

File: internal/api/registry/blobs_test.go [hunk 2]
```diff
@@ -382,7 +381,7 @@ func TestBlobStreamedAndChunkedUpload(t *testing.T) {
 					Body:         assert.ByteData(blob.Contents),
 					ExpectStatus: http.StatusAccepted,
 				}.Check(t, h)
-				uploadURL := resp.Header.Get("Location")
+				uploadURL := resp.Header.Get("Location") //nolint:govet
```
Comment (majewsky) [inline internal/api/registry/blobs_test.go:384]: Just out of curiosity, what was govet complaining about here?

**Review by majewsky — APPROVED** [review]
Thanks! One nitpick and one question, then this can go in.

File: internal/api/registry/blobs_test.go (hunk 2 above)
Comment (talal) [reply internal/api/registry/blobs_test.go:384]: Variable shadowing of `uploadURL`.

Comment (majewsky) [reply internal/api/registry/blobs_test.go:384]: I see, thanks!

**Pull request merged**